roles/servicetelemetry/tasks/component_grafana.yml

```diff
@@ -47,94 +26,247 @@
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: rhos-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: rhos-dashboard.json
-            json: |
-              {{ lookup('file', 'rhos-dashboard.json') | string }}
-
-    - name: Load Memcached Dashboard
-      k8s:
-        state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
-        definition:
-          apiVersion: integreatly.org/v1alpha1
-          kind: GrafanaDashboard
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: memcached-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: memcached-dashboard.json
-            json: |
-              {{ lookup('file', 'memcached-dashboard.json') | string }}
-
-    - name: Load Virtual Machine View Dashboard
-      k8s:
-        state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
-        definition:
-          apiVersion: integreatly.org/v1alpha1
-          kind: GrafanaDashboard
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: virtual-machine-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: virtual-machine-view.json
-            json: |
-              {{ lookup('file', 'virtual-machine-view.json') | string }}
+#---- deploy Grafana with v5 Operator (preferred)
+    - when: has_grafana_integreatly_api
+      name: Deploying with Grafana Operator v5
+      block:
+        - name: Construct oauth redirect reference
+          set_fact:
+            grafana_oauth_redir_ref:
+              kind: OAuthRedirectReference
+              apiVersion: v1
+              reference:
+                kind: Route
+                name: '{{ ansible_operator_meta.name }}-grafana-route'
+
+        - name: Lookup template
+          debug:
+            msg: "{{ lookup('template', './manifest_grafana_v5.j2') | from_yaml }}"
+
+        - name: Set default Grafana manifest (Grafana Operator v5)
+          set_fact:
+            grafana_manifest: "{{ lookup('template', './manifest_grafana_v5.j2') | from_yaml }}"
+          when: grafana_manifest is not defined
+
+        - name: Create an instance of Grafana (Grafana Operator v5)
+          k8s:
+            state: '{{ "present" if servicetelemetry_vars.graphing.enabled else "absent" }}'
+            definition:
+              '{{ grafana_manifest }}'
+
+        # NOTE: we only provide events forwarding with STF. We don't use events
+        # in dashboards, so there is no need to create an Elasticsearch
+        # datasource.
```

Comment:
I took some liberties here and only implemented Prometheus as a datasource for Grafana Operator v5. In Grafana Operator v4 the datasources were in a single datasource configuration as a list, but that's not the case here.

I was tempted to also remove the Elasticsearch datasource from the Grafana Operator v4 implementation, but thought that to be out of scope. Could certainly look into do that after this though.